### ISSUE TITLE
ClientUpdateException ArgumentNullException / Crash AggregateException (V0.7.3.0)

### ISSUE THREAD

**Issue by Voosle** [StrangeLoopGames/EcoIssues#6603]
Loading a saved game failed with V0.7.3.0 (also with version 0.7.2.5)
May I provide the storage files?

[Crash AggregateException 03295126.zip](https://github.com/StrangeLoopGames/EcoIssues/files/1858063/Crash.AggregateException.03295126.zip)

Save game taken from server where ClientUpdateException ArgumentNullException occured:
[ClientUpdateException ArgumentNullException 03283258.zip](https://github.com/StrangeLoopGames/EcoIssues/files/1858096/ClientUpdateException.ArgumentNullException.03283258.zip)

**Comment by MarjieAVolk** [comment]
Hm yeah one of your world objects can't load.  Yes, your save files would be useful, thanks.  Do you know where they are?

**Comment by Voosle** [comment]
File ist to large for GitHub. 
I hope this will work: https://1drv.ms/f/s!AovZo2hpobmim81iqPHJ6yd5yqkQHg

**Comment by MarjieAVolk** [comment]
Okay, I'm seeing the error message "'Unable to cast object of type 'Eco.Mods.TechTree.DirtRoadBlock' to type 'Eco.Gameplay.Objects.WorldObject'.'", while trying to load a Door  >>  I am not sure why, but anyways I tried deleting the offending door from the save, and got the same message for a different door.  I tried deleting all doors from the save, and now the save works.

So here is the doctored save: https://drive.google.com/open?id=1H-jp2VguXd9DK-iopgAbT7RXqTaU_S6O  All doors in the world have been deleted, but otherwise intact :P

**Comment by MarjieAVolk** [comment]
What previous version(s) was this save file on?  Was it ever run on a staging version?

**Comment by Voosle** [comment]
Hi,
are you sure, you put the right file to that link?
With that files I got the same crash and the file size is equal to my provided file.

**Comment by MarjieAVolk** [comment]
Oh hm I guess it is not the right one.  Sorry, let me update that.

**Comment by MarjieAVolk** [comment]
Okay, try now (same link https://drive.google.com/open?id=1H-jp2VguXd9DK-iopgAbT7RXqTaU_S6O)

**Comment by Voosle** [comment]
We used this world on a server running V0.7.2.5. Yesterday night (germany) the server recognized a new version 0.7.3.0 from steam, downloaded it and the worlds works fine. We switched to the 7.3 client and played for som hours.
Then one player recognized that steam was revert to 7.2.5 with the public branch. We started a backup of the server. That causes the game server to stop and next a new (old) version was installed again on our server (hosted at a gaming server provider in germany).
So, in conclusion: We never switched to a staging version by our selfes. But the confusing "up-and down-dates" may result in a slot when the saved game was converted to a staging version... :-)

**Comment by MarjieAVolk** [comment]
Ah, going down a version is not usually supported, so that is probably related.

**Comment by Voosle** [comment]
o.k. I supposed something like that :-)
The fixed save game works localy on my machine. I'm going to upload it to our server now.

**Comment by MarjieAVolk** [comment]
You may want to have a door redistribution party :P

**Comment by MarjieAVolk** [comment]
Guess I will close this, assuming it was related to the accidental early release of 7.3.  Thanks!

**Comment by Voosle** [comment]
Yes, closing is o.k.
I can't confim that it runs on our server, but that's because of V7.2.5. Don't know why the server is not updating as configured. I opened a ticket there :-)

Thank you very much for you quick support!
Great game!
